src/Middleware/HttpsPolicy/src/HttpsRedirectionMiddleware.cs

```diff
@@ -44,8 +41,14 @@ public HttpsRedirectionMiddleware(RequestDelegate next, IOptions<HttpsRedirectio
                 throw new ArgumentNullException(nameof(options));
             }
             var httpsRedirectionOptions = options.Value;
-            _httpsPort = httpsRedirectionOptions.HttpsPort;
-            _portEvaluated = _httpsPort.HasValue;
+            if (httpsRedirectionOptions.HttpsPort.HasValue)
+            {
+                _httpsPort = new Lazy<int>(() => httpsRedirectionOptions.HttpsPort.Value);
```

Comment:
`new Lazy<int>(httpsRedirectionOptions.HttpsPort.Value)` should work

Reply:
I don't think it does. The only methods that take T are funcs.

Reply:
https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/Lazy.cs#L219

Reply:
It's in netstandard2.1, this package targets 2.0. I can probably change it in 2.2/3.0